pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/gates/tests/Test_TNCuda_NonParam.cpp

```diff
@@ -0,0 +1,928 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the License);
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an AS IS BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <complex>
+#include <vector>
+
+#include <catch2/catch.hpp>
+
+#include "DevTag.hpp"
+#include "ExaTNCuda.hpp"
+#include "MPSTNCuda.hpp"
+#include "TNCudaGateCache.hpp"
+
+#include "TestHelpers.hpp"
+
+/// @cond DEV
+namespace {
+using namespace Pennylane::LightningGPU;
+using namespace Pennylane::LightningTensor;
+using namespace Pennylane::LightningTensor::TNCuda::Gates;
+using namespace Pennylane::Util;
+namespace cuUtil = Pennylane::LightningGPU::Util;
+using TestTNBackends =
+    Pennylane::Util::TypeList<MPSTNCuda<float>, MPSTNCuda<double>,
+                              ExaTNCuda<float>, ExaTNCuda<double>>;
+} // namespace
+/// @endcond
+
+TEMPLATE_LIST_TEST_CASE("TNCuda::Gates::Identity", "[TNCuda_Nonparam]",
+                        TestTNBackends) {
+    const bool inverse = GENERATE(false, true);
+    using TNDevice_T = TestType;
+    using cp_t = typename TNDevice_T::ComplexT;
+    std::size_t num_qubits = 3;
+    std::size_t maxExtent = 2;
+    DevTag<int> dev_tag{0, 0};
+
+    std::unique_ptr<TNDevice_T> tn_state;
+
+    if constexpr (std::is_same_v<TestType, MPSTNCuda<double>> ||
+                  std::is_same_v<TestType, MPSTNCuda<float>>) {
+        // Create the object for MPSTNCuda
+        tn_state = std::make_unique<TNDevice_T>(num_qubits, maxExtent, dev_tag);
+    } else {
+        // Create the object for ExaTNCuda
+        tn_state = std::make_unique<TNDevice_T>(num_qubits, dev_tag);
+    }
```

Comment:
Could we move these lines to a helper functor?

Reply:
Done